ext/dom/php_dom.stub.php

```diff
@@ -1,17 +1,55 @@
 <?php
 
-/** @generate-function-entries */
+/**
+ * @generate-function-entries
+ * @generate-class-entries
+ */
 
-class DOMDocumentType
+class DOMDocumentType extends DOMNode
 {
+    /**
+     * @dynamic
+     * @readonly
+     */
+    public string $name;
```

Comment:
I'm not sure these properties should be here. Stubs are supposed to be the source of truth, and these properties are lies (until such a time as we support property accessors).

Reply:
I somewhat disagree here. While php-src itself unfortunately can't yet use these properties as the source of truth, static analysis tools would. If we add normal properties to stubs, I think we should also add dynamic ones as well: especially because maintenance would become easier if we could at least auto-generate property definitions in the manual based on the the stubs. I've already found out that the manual lies about the types in quite a few cases, and misses some of the properties.

Or adding dynamic properties still doesn't make sense for you? If not, I'll get rid of them.

Reply:
I'm not sure about this. For the argument/return stubs we have tried to make sure that everything we annotate follows standard semantics, but here we have for example

```
    /** @dynamic */
    public ?string $nodeValue;
```

where the setter will simply convert the value to string, not following strict_types. I think documenting it this way is problematic.

Reply:
Yeah, I was unsure whether I should use phpdoc-style types or not for a while, but finally, I went with the native types everywhere. I agree, this was a mistake (I mixed up the assignment rules). So would using phpdoc type annotation make sense for you for read-write dynamic properties?

If you have concerns about these internal dynamic properties, I can apply the rest of the changes only, and keep the problematic ones until you have a clear decision. :)

Reply:
Yeah, I think it would make sense to apply the other changes first (those all look good to me).

Reply:
I've just applied everything else, and rebased this PR. Let's leave it open until the decision is made.